core/src/main/java/org/bitcoinj/wallet/WalletFiles.java

```diff
@@ -94,10 +94,11 @@ public WalletFiles(final Wallet wallet, File file, Duration delay) {
                 // Some other scheduled request already beat us to it.
                 return null;
             }
-            Date lastBlockSeenTime = wallet.getLastBlockSeenTime();
             log.info("Background saving wallet; last seen block is height {}, date {}, hash {}",
                     wallet.getLastBlockSeenHeight(),
-                    lastBlockSeenTime != null ? TimeUtils.dateTimeFormat(lastBlockSeenTime) : "unknown",
+                    wallet.getLastBlockSeenTimeInstant()
+                            .map(time -> TimeUtils.dateTimeFormat(time.toEpochMilli()))
```

Comment:
This could be a method reference once we add `TimeUtils::dateTimeFormat` that takes an `Instant`.

Reply:
I'll keep it in mind. The dateTimeFormat modernization is close to finished.